core/templates/pages/about-page/about-page.controller.spec.ts

```diff
@@ -0,0 +1,221 @@
+    // finish. setTimeout is executed only after call stack is empty.
+    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
+    setTimeout(function() {
+      expect(windowRef.nativeWindow.location.hash).toBe('#about');
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(addClassSpy).toHaveBeenCalledTimes(4);
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(removeClassSpy).toHaveBeenCalledTimes(4);
+      done();
+    });
+  });
+
+  it('should click on license tab', function(done) {
+    // @ts-ignore
+    var addClassSpy = spyOn($.fn, 'addClass').and.callThrough();
+    // @ts-ignore
+    var removeClassSpy = spyOn($.fn, 'removeClass').and.callThrough();
+
+    ctrl.$onInit();
+    ctrl.onTabClick('license');
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      location: {
+        hash: '#license',
+        reload: function() {}
+      }
+    });
+
+    // setTimeout is being used here in order to wait onhashchange event to
+    // finish. setTimeout is executed only after call stack is empty.
+    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
+    setTimeout(function() {
+      expect(windowRef.nativeWindow.location.hash).toBe('#license');
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(addClassSpy).toHaveBeenCalledTimes(4);
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(removeClassSpy).toHaveBeenCalledTimes(4);
+      done();
+    });
+  });
+
+  it('should click on foundation tab', function(done) {
+    // @ts-ignore
+    var addClassSpy = spyOn($.fn, 'addClass').and.callThrough();
+    // @ts-ignore
+    var removeClassSpy = spyOn($.fn, 'removeClass').and.callThrough();
+
+    ctrl.$onInit();
+    ctrl.onTabClick('foundation');
+
+    // setTimeout is being used here in order to wait onhashchange event to
+    // finish. setTimeout is executed only after call stack is empty.
+    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
+    setTimeout(function() {
+      expect(windowRef.nativeWindow.location.hash).toBe('#foundation');
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(addClassSpy).toHaveBeenCalledTimes(4);
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(removeClassSpy).toHaveBeenCalledTimes(4);
+      done();
+    });
+  });
+
+  it('should click on credits tab', function(done) {
+    // @ts-ignore
+    var addClassSpy = spyOn($.fn, 'addClass').and.callThrough();
+    // @ts-ignore
+    var removeClassSpy = spyOn($.fn, 'removeClass').and.callThrough();
+
+    ctrl.$onInit();
+    ctrl.onTabClick('credits');
```

Comment:
I agree with you but license tab has a different behavior.

Reply:
Then do the consolidation only for the ones that have the same behaviour.

Reply:
I have a question: is it not better to keep these tests separately? If one of the tabs changes its behavior in the future, it would be necessary to restructure the tests again.

Reply:
Good question. It depends on how likely the behaviour is to change. Given that the behaviour hasn't changed in quite a long time -- it probably is fine to consolidate?

Reply:
License is different because it calls `window.reload` so the test needs to be careful so the `reload` won't be called during the tests, otherwise it would fail.

Reply:
I'm not sure if it's a good idea to wrap it together. I tried to do it and I faced some errors related to `addClassSpy` and `removeClassSpy` spies. I can't spy more than one time and I don't know how to reset the spy after each iteration since it considers the total times. So in the last iteration, both spies should have been called 12 times (not considering license tab on here). I think it might confuse who is going to write this code.

Reply:
OK, I'll defer to your decision here. Let's keep it as you have it. Thanks!